readchar/_posix_read.py

```diff
@@ -19,7 +19,10 @@ def readchar() -> str:
         term[3] &= ~(termios.ICANON | termios.ECHO | termios.IGNBRK | termios.BRKINT)
         termios.tcsetattr(fd, termios.TCSAFLUSH, term)
 
-        ch = sys.stdin.read(1)
+        if sys.platform == "darwin":
+            ch = sys.stdin.readline(1)
+        else:
+            ch = sys.stdin.read(1)
```

Comment:
this is defnetly to complex for me to just sign of on. Please remove #70 from this PR. I will decide if it makes sense to include it once I finde the time to get around to it.